### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by dweindl**
* Const
* override
* Cleanup
* First step towards #179
...

### CODE REVIEW

File: include/rdata.h [hunk 1]
```diff
@@ -1,26 +1,23 @@
 #ifndef _MY_RDATA
 #define _MY_RDATA
 #include <include/udata.h>
-class UserData;
+
 class Model;
```
Comment (FFroehlich) [inline include/rdata.h]: stray class Model? was removed in udata.h

File: include/udata.h [hunk 2]
```diff
@@ -4,129 +4,187 @@
...
-    /** internal sensitivity method */
-    /*!
-     * a flag used to select the sensitivity solution method. Its value can be
-     * CV SIMULTANEOUS or CV STAGGERED. Only applies for Forward Sensitivities.
-     */
-    int ism;
+    /** internal sensitivity method flag used to select the sensitivity solution
+     * method. Its value can be CV SIMULTANEOUS or CV STAGGERED. Only applies
+     * for Forward Sensitivities. */
+    int ism = 1;
```
Comment (FFroehlich) [inline include/udata.h]: please use CV_SIMULTANEOUS here (from cvodes.h, or redefine as amici constant?)

File: include/udata.h [hunk 3]
```diff
@@ -4,129 +4,187 @@
...
 
     /** method for sensitivity computation */
-    AMICI_sensi_meth sensi_meth;
+    AMICI_sensi_meth sensi_meth = AMICI_SENSI_FSA;
 
     /** linear solver specification */
-    int linsol;
+    int linsol = 9;
```
Comment (FFroehlich) [inline include/udata.h]: AMICI_KLU

File: include/udata.h [hunk 4]
```diff
@@ -4,129 +4,187 @@
...
 
-    /** interpolation type */
-    /*!
-     * specifies the interpolation type for the forward problem solution which
+    /** interpolation type for the forward problem solution which
      * is then used for the backwards problem. can be either CV_POLYNOMIAL or
      * CV_HERMITE
      */
-    int interpType;
+    int interpType = 1;
```
Comment (FFroehlich) [inline include/udata.h]: CV_HERMITE

File: include/udata.h [hunk 5]
```diff
@@ -4,129 +4,187 @@
...
 
-    /** linear multistep method */
-    /*!
-     * specifies the linear multistep method and may be one of two possible
+    /** specifies the linear multistep method and may be one of two possible
      * values: CV ADAMS or CV BDF.
      */
-    int lmm;
+    int lmm = 2;
```
Comment (FFroehlich) [inline include/udata.h]: CV_BDF

File: include/udata.h [hunk 6]
```diff
@@ -4,129 +4,187 @@
...
 
-    /** nonlinear solver */
-    /*!
+    /**
      * specifies the type of nonlinear solver iteration and may be either CV
      * NEWTON or CV FUNCTIONAL.
      */
-    int iter;
+    int iter = 2;
```
Comment (FFroehlich) [inline include/udata.h]: CV_NEWTON

File: include/udata.h [hunk 7]
```diff
@@ -4,129 +4,187 @@
...
-    double *sx0data;
+    /** sensitivity initialisation (size nx * nplist) */
+    double *sx0data = nullptr;
 
     /** state ordering */
-    int ordering;
-    
-    /** function to print the contents of the UserData object */
-    void print();
+    int ordering = 0;
```
Comment (FFroehlich) [inline include/udata.h]: options are AMD: 0 COLAMD: 1 natural:2 I don't think we have corresponding defines yet

File: src/amici_interface_matlab.cpp [hunk 8]
```diff
@@ -19,19 +19,38 @@
...
+    RHS_CONSTANTS,
+    RHS_OPTIONS,
+    RHS_PLIST,
+    RHS_PBAR,
+    RHS_XSCALE_UNUSED,
+    RHS_INITIALIZATION,
+    RHS_DATA,
+    RHS_NUMARGS_REQUIRED = RHS_DATA,
+    RHS_NUMARGS
+};
```
Comment (FFroehlich) [inline src/amici_interface_matlab.cpp]: 👍

File: src/forwardproblem.cpp [hunk 9]
```diff
@@ -470,15 +481,19 @@ int ForwardProblem::getEventOutput(UserData *udata, ReturnData *rdata,
     return status;
 }
 
-/* -------------------------------------------------------------------------------- */
-/* -------------------------------------------------------------------------------- */
-/* -------------------------------------------------------------------------------- */
+/* --------------------------------------------------------------------------------
```
Comment (FFroehlich) [inline src/forwardproblem.cpp]: please fix by removing a couple of `-`

File: include/rdata.h (hunk 1 above)
Comment (dweindl) [reply include/rdata.h]: We need it for `    ReturnData(const UserData *udata, const Model *model);
`

File: src/forwardproblem.cpp (hunk 9 above)
Comment (dweindl) [reply src/forwardproblem.cpp]: All done.

**Pull request merged**